File: 884 - Uncommon Words from Two Sentences.py
class Solution:
    def uncommonFromSentences(self, A: str, B: str) -> list[str]:
        a = A.split(' ')
        aset = set(a)
        b = B.split(' ')
        bset = set(b)
        uncommon = []
        for word in a:
            if word not in bset:
                uncommon.append(word)
        for word in b:
            if word not in aset:
                uncommon.append(word)
        rc = dict()
        for word in uncommon:
            rc[word] = rc[word] + 1 if rc.get(word) is not None else 1
        res = []
        for word in rc.keys():
            if rc[word] == 1:
                res.append(word)
        return res

File: 884 - Uncommon Words from Two Sentences.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;

import org.graalvm.compiler.replacements.nodes.ReadRegisterNode;

class Solution {
    public String[] uncommonFromSentences(String A, String B) {
        Map<String, Integer> count = new HashMap<>();
        for (String word : A.split(" ")) {
            count.put(word, count.getOrDefault(word, 0) + 1);
        }
        for (String word : B.split(" ")) {
            count.put(word, count.getOrDefault(word, 0) + 1);
        }
        List<String> res = new ArrayList<>();
        for (Entry<String, Integer> e : count.entrySet()) {
            if (e.getValue() == 1) {
                res.add(e.getKey());
            }
        }
        return res.toArray(new String[res.size()]);
    }
}